This patch is a simplified version of a patch I have sent upstream.
http://gerrit.rockbox.org/1825

-- Chewi

diff -Naur a/base/ziputil.cpp b/base/ziputil.cpp
--- a/base/ziputil.cpp	2014-01-05 22:44:53.000000000 +0000
+++ b/base/ziputil.cpp	2018-02-25 10:55:36.994002648 +0000
@@ -22,9 +22,9 @@
 #include "progressloggerinterface.h"
 #include "Logger.h"
 
-#include "quazip/quazip.h"
-#include "quazip/quazipfile.h"
-#include "quazip/quazipfileinfo.h"
+#include <quazip5/quazip.h>
+#include <quazip5/quazipfile.h>
+#include <quazip5/quazipfileinfo.h>
 
 
 ZipUtil::ZipUtil(QObject* parent) : ArchiveUtil(parent)
diff -Naur a/base/ziputil.h b/base/ziputil.h
--- a/base/ziputil.h	2014-01-05 22:44:54.000000000 +0000
+++ b/base/ziputil.h	2018-02-25 10:55:28.513838004 +0000
@@ -21,9 +21,9 @@
 
 #include <QtCore>
 #include "archiveutil.h"
-#include "quazip/quazip.h"
-#include "quazip/quazipfile.h"
-#include "quazip/quazipfileinfo.h"
+#include <quazip5/quazip.h>
+#include <quazip5/quazipfile.h>
+#include <quazip5/quazipfileinfo.h>
 
 class ZipUtil : public ArchiveUtil
 {
diff -Naur a/rbutilqt.pri b/rbutilqt.pri
--- a/rbutilqt.pri	2014-01-05 22:44:57.000000000 +0000
+++ b/rbutilqt.pri	2018-02-25 10:54:39.793852035 +0000
@@ -69,12 +69,6 @@
  base/rockboxinfo.cpp \
  ../../tools/mkboot.c \
  ../../tools/iriver.c \
- quazip/quazip.cpp \
- quazip/quazipfile.cpp \
- quazip/quazipnewinfo.cpp \
- quazip/unzip.c \
- quazip/zip.c \
- quazip/ioapi.c \
  base/ziputil.cpp \
  gui/comboboxviewdelegate.cpp \
  gui/selectiveinstallwidget.cpp \
@@ -156,14 +150,6 @@
  base/rockboxinfo.h \
  ../../tools/mkboot.h \
  ../../tools/iriver.h \
- quazip/crypt.h \
- quazip/ioapi.h \
- quazip/quazipfile.h \
- quazip/quazipfileinfo.h \
- quazip/quazip.h \
- quazip/quazipnewinfo.h \
- quazip/unzip.h \
- quazip/zip.h \
  base/ziputil.h \
  lame/lame.h \
  gui/comboboxviewdelegate.h \
diff -Naur a/rbutilqt.pro b/rbutilqt.pro
--- a/rbutilqt.pro	2014-01-05 22:45:00.000000000 +0000
+++ b/rbutilqt.pro	2018-02-25 10:55:04.577357613 +0000
@@ -114,7 +114,7 @@
 
 DEPENDPATH = $$INCLUDEPATH
 
-LIBS += -L$$OUT_PWD -L$$MYLIBBUILDDIR
+LIBS += -L$$OUT_PWD -L$$MYLIBBUILDDIR -lquazip5
 # append all RBLIBS to LIBS
 for(rblib, RBLIBS) {
     LIBS += -l$$rblib
@@ -122,7 +122,6 @@
 
 # on win32 libz is linked implicitly.
 !win32 {
-    LIBS += -lz
 }
 
 # Add a (possibly found) libspeex now, don't do this before -lrbspeex!
